### PR TITLE
Upgrade from MUI v4 to v5, React 16 to 18

### PR DESCRIPTION
**Pull request by andrewkfiedler**
A work in progress.  Putting up here so I can get a build to reference in downstream projects.

### CODE REVIEW

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/menu-state/menu-state.tsx [hunk 1]
```diff
@@ -130,9 +130,9 @@ export const useMenuState = ({ maxHeight }: Props = {}) => {
       >
     >,
     MuiButtonProps: {
-      innerRef: anchorRef,
+      ref: anchorRef as unknown as ButtonProps['ref'], // todo: see if we need to change type instead
```
Comment (sgalla) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/menu-state/menu-state.tsx]: Can this todo be removed?

Comment (jrnorth) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/menu-state/menu-state.tsx]: status of this todo?

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/pages/search.tsx [hunk 2]
```diff
@@ -369,8 +369,8 @@ export const OpenSearch = ({
           if (highlightedElementString) {
             setCurrentHighlight(
               (
-                document.querySelector(
-                  `#${highlightedElementString}`
+                document.getElementById(
+                  `${highlightedElementString}`
```
Comment (jrnorth) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/pages/search.tsx]: ✏️ Don't need the template string anymore.

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/pages/search.tsx [hunk 3]
```diff
@@ -1089,13 +1094,14 @@ const LeftTop = () => {
               </Paper>
             </Popover>
             <Button
+              color="inherit"
               component="div"
-              className={`children-block children-h-full text-left text-2xl shrink truncate ${
+              className={` text-left text-2xl shrink truncate ${
```
Comment (jrnorth) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/pages/search.tsx]: ✏️ Extra leading space.

**Review by sgalla — APPROVED** [review]

**Review by sgalla — APPROVED** [review]

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/tabs/metacard/transfer-list.tsx [hunk 4]
```diff
@@ -178,7 +178,7 @@ const ItemRow = ({
       <Button
         fullWidth
         size="medium"
-        className="children-block text-left"
+        className=" justify-start"
```
Comment (jrnorth) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/tabs/metacard/transfer-list.tsx]: ✏️ Extra leading space.

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/tabs/metacard/transfer-list.tsx [hunk 5]
```diff
@@ -356,7 +356,7 @@ const CustomList = ({
               disabled={Object.keys(items).length === 0}
               onClick={handleToggleAll(items)}
               color={
-                isIndeterminate || isCompletelySelected ? 'default' : 'default'
+                isIndeterminate || isCompletelySelected ? 'primary' : 'primary'
```
Comment (jrnorth) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/tabs/metacard/transfer-list.tsx]: We can remove this ternary.

**Review by jrnorth — APPROVED** [review]

**Pull request merged**